prometheus/counter_test.go

```diff
@@ -273,3 +273,52 @@ func TestCounterExemplar(t *testing.T) {
 		t.Error("adding exemplar with oversized labels succeeded")
 	}
 }
+
+func TestCounterCreatedTimestamp(t *testing.T) {
+	now := time.Now()
+
+	counter := NewCounter(CounterOpts{
+		Name: "test",
+		Help: "test help",
+	}).(*counter)
+	counter.now = func() time.Time { return now }
+	counter.initCreated()
+
+	ts := timestamppb.New(now)
+	if err := ts.CheckValid(); err != nil {
+		t.Fatal(err)
+	}
+
+	expectedCounter := &dto.Counter{
+		CreatedTimestamp: ts,
+	}
+
+	if counter.createdTs.AsTime() != expectedCounter.CreatedTimestamp.AsTime() {
+		t.Errorf("expected created timestamp %s, got %s", expectedCounter.CreatedTimestamp, counter.createdTs)
+	}
+}
+
+func TestCounterVecCreatedTimestamp(t *testing.T) {
+	now := time.Now()
+
+	counterVec := NewCounterVec(CounterOpts{
+		Name: "test",
+		Help: "test help",
+	}, []string{"label"})
+	counter := counterVec.WithLabelValues("value").(*counter)
+	counter.now = func() time.Time { return now }
+	counter.initCreated()
+
+	ts := timestamppb.New(now)
+	if err := ts.CheckValid(); err != nil {
+		t.Fatal(err)
+	}
+
+	expectedCounter := &dto.Counter{
+		CreatedTimestamp: ts,
+	}
+
+	if counter.createdTs.AsTime() != expectedCounter.CreatedTimestamp.AsTime() {
```

Comment:
Can we instead use counter.Write (or some form of it), and check if resulted DTO has CT set.